server/api/api/endpoints/nuclio.py

```diff
@@ -587,6 +634,25 @@ def _handle_nuclio_deploy_status(
     )
 
 
+async def _get_api_gateways_hosts_for_function(
+    auth_info, project, name, tag
+) -> list[str]:
+    function_uri = generate_object_uri(project, name, tag)
+    async with server.api.utils.clients.async_nuclio.Client(auth_info) as client:
+        api_gateways = await client.list_api_gateways(project)
+        # if there are any API gateways, filter the ones associated with the function
+        # extract the hosts from the API gateway specifications and return them as a list
+        return (
+            [
+                api_gateway.spec.host
+                for api_gateway in api_gateways.values()
+                if function_uri in api_gateway.get_function_names()
+            ]
+            if api_gateways
+            else []
+        )
```

Comment:
@TomerShor I agree that it would be much better than filtering on mlrun side, but the only possible way to do it on Nuclio will require changes from Nuclio as well meaning either filtering by function name or by annotation contain given string. Currently we support only filtering by labels in Nuclio.

Reply:
Let's add a TODO then because we would want the optimized way.